### ISSUE TITLE
复现结果没有达到预期

### ISSUE THREAD

**Issue by WangZZJJ** [HorizonRobotics/Sparse4D#76]
林博您好，我做了几组实验但是不能得到预期的结果，以下是我训练的log和结果。以下实验均是用8张A100训练。
1. 我用配置sparse4dv3_temporal_r50_1x8_bs6_256x704训练，一开始loss下降，后来grad_norm出现nan并且报错。这是我的log
[20240722_143807.log](https://github.com/user-attachments/files/16346150/20240722_143807.log)
2. 然后我调小了学习率为6e-5，调整了epoch数为150。得到的训练结果NDS小了3个点。下面是我的结果和log
[20240719_075541.log](https://github.com/user-attachments/files/16346181/20240719_075541.log)
![image](https://github.com/user-attachments/assets/42dd02f3-686e-4c27-a7f7-e8680ab841e8)
3. 接着我修改了，lr=1e-4和"img_backbone": dict(lr_mult=0.2)。得到的结果NDS还是比预期小了3个点。
[20240722_032236.log](https://github.com/user-attachments/files/16346228/20240722_032236.log) 
![image](https://github.com/user-attachments/assets/ff787b3f-5e58-4841-b9b6-3bd1dbff0188)
请问，您有什么建议吗？感谢！

**Comment by linxuewu** [comment]
emmm，你这个指标确实差的比较远。

如果是8卡训练的话，确保代码版本、config、anchor_file、data pkl都和我提供的完全一致吗？

**Comment by mikasa-2024** [comment]
> 林博您好，我做了几组实验但是不能得到预期的结果，以下是我训练的log和结果。以下实验均是用8张A100训练。
> 
> 1. 我用配置sparse4dv3_temporal_r50_1x8_bs6_256x704训练，一开始loss下降，后来grad_norm出现nan并且报错。这是我的log
>    [20240722_143807.log](https://github.com/user-attachments/files/16346150/20240722_143807.log)
> 2. 然后我调小了学习率为6e-5，调整了epoch数为150。得到的训练结果NDS小了3个点。下面是我的结果和log
>    [20240719_075541.log](https://github.com/user-attachments/files/16346181/20240719_075541.log)
>    ![image](https://private-user-images.githubusercontent.com/57397386/351256825-42dd02f3-686e-4c27-a7f7-e8680ab841e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyOTMxNjgsIm5iZiI6MTcyNDI5Mjg2OCwicGF0aCI6Ii81NzM5NzM4Ni8zNTEyNTY4MjUtNDJkZDAyZjMtNjg2ZS00YzI3LWE3ZjctZTg2ODBhYjg0MWU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDAyMTQyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1YjAwYmU2YzNhNmFkYWMxZThiZDU0YTI0MWYwYWEyMGFmZjlhN2ViM2ZhNDFhYTkyNzcyNDMxZTM4NjI1NmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0CdX6JLu6vHvrRnBr_F9hd-kOqPA7vzf6ubsQtVl_EA)
> 3. 接着我修改了，lr=1e-4和"img_backbone": dict(lr_mult=0.2)。得到的结果NDS还是比预期小了3个点。
>    [20240722_032236.log](https://github.com/user-attachments/files/16346228/20240722_032236.log)
>    ![image](https://private-user-images.githubusercontent.com/57397386/351257481-ff787b3f-5e58-4841-b9b6-3bd1dbff0188.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyOTMxNjgsIm5iZiI6MTcyNDI5Mjg2OCwicGF0aCI6Ii81NzM5NzM4Ni8zNTEyNTc0ODEtZmY3ODdiM2YtNWU1OC00ODQxLWI5YjYtM2JkMWRiZmYwMTg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDAyMTQyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxYTg3YWNhMDMyZDBjNmU1YjAyMDc5ZjE3MTlkNWExNmY3MWFlOTRhZjM1MWY0MGNjNzNjYjdiYjgwNTRhOGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-_9YOKnVR1LK0SB2RHIaJIFcj796XXFihhUQqb7vYc8)
>    请问，您有什么建议吗？感谢！

哈咯，请问你解决了吗，我也是用8张A100训练中会出现梯度爆炸

**Comment by PonyAIjkz** [comment]
@mikasa-2024 @WangZZJJ 由于代码在处理3D 尺寸过程中使用exp, 所以预测尺寸的冪很容易爆炸,建议使用较小的学习率,并且关闭训练epoch后推理的hook机制,可以复现指标

**Comment by mikasa-2024** [comment]
> @mikasa-2024 @WangZZJJ 由于代码在处理3D 尺寸过程中使用exp, 所以预测尺寸的冪很容易爆炸,建议使用较小的学习率,并且关闭训练epoch后推理的hook机制,可以复现指标

请问‘关闭训练epoch后推理的hook机制’是指什么，没有太明白，意思是训练中途不进行评测，仅在全部训练完成后评测一次吗？

**Comment by PonyAIjkz** [comment]
是的, 前期训练不稳定也不容易收敛, 评测可以放在后面离线进行, 我记得训练过程中加上--no-validate这个参数就可以了

**Comment by mikasa-2024** [comment]
> 是的, 前期训练不稳定也不容易收敛, 评测可以放在后面离线进行, 我记得训练过程中加上--no-validate这个参数就可以了

好的，我之后尝试一下，感谢！
